src/action.js

```diff
@@ -292,7 +292,7 @@ export function dndzone(node, options) {
         shadowElData = {...draggedElData, isDndShadowItem: true};
 
         // creating the draggable element
-        draggedEl = createDraggedElementFrom(originalDragTarget);
+        draggedEl = createDraggedElementFrom(originalDragTarget, config.transformDraggedElement, currentIdx);
```

Comment:
1. This is the config of the dnd zone it is dragged from. shouldn't it be manipulated based on the config of the dnd zone it enters?
2. What are you using the origin index for in the transform function?

Reply:
I originally had it in the morph function, but it was being called for every single element that the draggable was going over and not being called for the last element in the list of items.

I am using the index to tell which item I dragged so I am able to change what the draggable looked like.

For instance:

```
[
      { id: randomString(), name: "string" },
      { id: randomString(), name: "combobox" },
      { id: randomString(), name: "spacer" },
    ]
```

If my items looks like this, I can check if index === 1 and then change the dragged element html to look like a combobox, etc.

Reply:
Here is how it functions now. I am able to use the index to change the draggable accordingly.

https://streamable.com/zca69w

Reply:
I am glad it works in this case but we need something that works generically for basically every situation the library can find itself in. Not sure why u had issues with the last item and such. I can probably help but...
Please help me understand the desired behaviour first. Feel free to correct any mistake I am making below. My understanding is:
- We want to define a function **per dndzone** and pass it in as config (as in dndzone1 might have f1 and dndzone2 might have f2)
- We expect the function to be called whenever - here I am not super clear. Here are some options, more than one may apply:
a. when the dragged element enters the dndzone (f1 if it enters dndzone1, f2 for dndzone2)
b. when the dragged element hovers over a new index after it had entered
c. when the dragged element leaves a dndzone (is there a reverse function?)
d. when drag start and the element is created? if so, it might not be consistent with the current intention of morphing - to make the dragged element look like it "belongs" in its would be destination
e. ?

Reply:
d. is the use case I was going for with it. This way you can have full control of how it looks when it is being dragged if needed. The current way of just cloning the node and changing styles is perfect if you just want to move the same "item" from one place to another, but it is lacking in terms of being able to customize that to display something completely different.

My original issue where the text was being resized because of the morph is a perfect example of this. The sidebar is small and you wouldn't want to display a textbox on it, but a preview of how it would look before dropped on the page is still needed.

This change doesn't restrict or cause any additional concerns that you would need to consider, as all it is doing is opening it up to the developer to apply additional styles to the draggable element as they please.

Reply:
Transforming it when drag starts assumes there is only one other way it would need to look like, doesn't it? as in - it is assuming its destination and it is assuming the user doesn't intend to drop it back in its original zone

Reply:
Correct, even if the user does drop it back in its original dropzone, it doesn't keep those styles because it'll re-render with the updated item list.

Reply:
I think this is too specific of a usecase. 
Let's take a step back for a minute. 
is what you really want is what you did initially but u changed to the current approach because you encountered some issues?

Reply:
I see now what you mean. It would make more sense for the zone that is being dropped onto to decide how it should change. It does make more sense in the morph function. I confused myself before by putting the function on the wrong dropzone.

I moved it to the morph function and it works well for transforming. The only thing I need is the index of the dragged item so I know what to transform it to.

Reply:
maybe it can pass in the data of the item `draggedElData`?

Reply:
Yeah something like that would work, i'll take a look at it later. Do you plan on exposing any additional events such as onDragStart, onDragStop? From what I recall, consider and finalize are only invoked once you are over the dropzone.

Reply:
consider is invoked on every change except for the final drop that invokes finalize. it doesn't include any information on why it was invoked but we could add it if needed (while trying to keep things as simple as possible)